AsyncDisplayKit/ASTableView.mm

```diff
@@ -1584,6 +1596,31 @@ - (void)clearFetchedData
 
 #pragma mark - Helper Methods
 
+// Note: This is called every layout, and so it is very performance sensitive.
+- (CGFloat)sectionIndexWidth
+{
+  // If they don't implement the methods, then there's no section index.
+  if (_asyncDataSourceFlags.sectionIndexMethods == NO) {
+    return 0;
+  }
+
+  UIView *indexView = _sectionIndexView;
+  if (indexView.superview == self) {
+    return indexView.frame.size.width;
+  }
+
+  CGRect bounds = self.bounds;
+  for (UIView *view in self.subviews) {
+    CGRect frame = view.frame;
+    // Section index is right-aligned and less than half-width.
+    if (CGRectGetMaxX(frame) == CGRectGetMaxX(bounds) && frame.size.width * 2 < bounds.size.width) {
```

Comment:
This seems pretty questionable... what about other subviews people may add to a table view, and manage with scrollViewDidScroll?

I think in the future, it would be more ideal to call super layoutSubviews in the layout method and then capture the bounds on a .contentView inside an _ASTableViewCell, which is sized to reflect these things - as well as delete control, resize handle, etc.  This would be tricky to say the least, but would be more UIKit-compliant if possible.